src/components/CompareResults/ResultsView.tsx

```diff
@@ -34,6 +45,21 @@ function ResultsView(props: ResultsViewProps) {
     document.title = title;
   }, [title]);
 
+  useEffect(() => {
+    const waitForData = async () => {
```

Comment:
add `setLoading(true)` before the `await` and I believe you can get rid of `handleLoaderRefresh`.
It shouldn't do a "double render" at load time because the value is already `true`.

Reply:
This is removed and no longer needed.